pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -0,0 +1,286 @@
+package openflow
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/contiv/libOpenflow/openflow13"
+	"github.com/contiv/ofnet/ofctrl"
+	"k8s.io/klog"
+)
+
+const (
+	OVSRunDir          = "/var/run/openvswitch"
+	ofTableExistsError = "Table already exists"
+)
+
+// ofTable implements openflow.Table.
+type ofTable struct {
+	sync.Mutex
+	id         TableIDType
+	next       TableIDType
+	missAction MissActionType
+	flowCount  uint
+	updateTime time.Time
+
+	*ofctrl.Table
+}
+
+func (t *ofTable) GetID() TableIDType {
+	return t.id
+}
+
+func (t *ofTable) Status() TableStatus {
+	return TableStatus{
+		ID:         uint(t.id),
+		FlowCount:  t.flowCount,
+		UpdateTime: t.updateTime,
+	}
+}
+
+func (t *ofTable) GetMissAction() MissActionType {
+	return t.missAction
+}
+
+func (t *ofTable) GetNext() TableIDType {
+	return t.next
+}
+
+func (t *ofTable) UpdateStatus(flowCountDelta int) {
+	t.Lock()
+	defer t.Unlock()
+
+	if flowCountDelta < 0 {
+		t.flowCount -= uint(-flowCountDelta)
+	} else {
+		t.flowCount += uint(flowCountDelta)
+	}
+	t.updateTime = time.Now()
+}
+
+// BuildFlow returns FlowBuilder object to help construct Openflow entry.
+func (t *ofTable) BuildFlow(priority uint16) FlowBuilder {
+	fb := new(ofFlowBuilder)
+	fb.table = t
+	// Set ofctl.Table to Flow, otherwise the flow can't find OFSwitch to install.
+	fb.Flow = ofctrl.Flow{Table: t.Table, Match: ofctrl.FlowMatch{Priority: priority}}
+	return fb
+}
+
+// DumpFlows dumps all existent Openflow entries from OFSwitch using cookie ID and table ID as filters
+func (t *ofTable) DumpFlows(cookieID, cookieMask uint64) map[uint64]*FlowStats {
+	ofStats := t.Table.Switch.DumpFlowStats(cookieID, cookieMask, nil, &t.TableId)
+	if ofStats == nil {
+		return nil
+	}
+	flowStats := make(map[uint64]*FlowStats)
+	for _, stat := range ofStats {
+		cookie := stat.Cookie
+		s := &FlowStats{
+			TableID:         stat.TableId,
+			PacketCount:     stat.PacketCount,
+			DurationNSecond: stat.DurationNSec,
+		}
+		flowStats[cookie] = s
+	}
+	return flowStats
+}
+
+func newOFTable(id, next TableIDType, missAction MissActionType) *ofTable {
+	return &ofTable{
+		id:         id,
+		next:       next,
+		missAction: missAction,
+	}
+}
+
+// OFBridge implements openflow.Bridge.
+type OFBridge struct {
+	sync.Mutex
+	bridgeName string
+	// tableCache is used to cache ofTables
```

Comment:
Could you add trailing period consistently to comments in this file?

Reply:
Done.